### ISSUE TITLE
can't build php56 from aur using yaourt

### ISSUE THREAD

**Issue by shilka** [archlinuxfr/yaourt#226]
tldr: yaourt -S aur/php56 results in not fully built package, some extensions are missing after installation

description: after yaourt -S aur/php56 the package seems to be installed in system, as yaourt marks it  "installed". In https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=php56 there are a lot of lines like

(example)
install -D -m755 ${srcdir}/build-php/modules/gd.so ${pkgdir}/usr/lib/${pkgbase}/modules/gd.so

but 
> ls /usr/lib/php56/modules/
resolves in
bcmath.so exif.so mysqli.so phar.so sysvmsg.so
bz2.so ftp.so mysql.so posix.so sysvsem.so
calendar.so gettext.so opcache.so shmop.so sysvshm.so
curl.so gmp.so openssl.so soap.so xmlrpc.so
dba.so iconv.so pdo_mysql.so sockets.so zip.so

which did not lists, for example, gd module

this resolves in troubles running php56
> php56
php56
PHP Warning: PHP Startup: Unable to load dynamic library '/usr/lib/php56/modules/gd.so' - /usr/lib/php56/modules/gd.so: cannot open shared object file: No such file or directory in Unknown on line 0
PHP Warning: PHP Startup: Unable to load dynamic library '/usr/lib/php56/modules/mcrypt.so' - /usr/lib/php56/modules/mcrypt.so: cannot open shared object file: No such file or directory in Unknown on line 0
PHP Warning: PHP Startup: Unable to load dynamic library '/usr/lib/php56/modules/pdo_pgsql.so' - /usr/lib/php56/modules/pdo_pgsql.so: cannot open shared object file: No such file or directory in Unknown on line 0
PHP Warning: PHP Startup: Unable to load dynamic library '/usr/lib/php56/modules/sqlite3.so' - /usr/lib/php56/modules/sqlite3.so: cannot open shared object file: No such file or directory in Unknown on line 0

Theese extensions (gd.so, mcrypt.so, pdo_pgsql.so, sqlite3.so) has not actually been built, but they should be. It's worth to mention that 

cower -d php56
cd php56
makepkg -si

Builds all package and corresponding extensions, that makes me think there is a possible bug in yaourt.

**Comment by f2404** [comment]
Hmm, can you provide full build log (preferably via Gist)?

**Comment by shilka** [comment]
Where can i find build log file? I have
[2016-02-03 10:27] [PACMAN] Running 'pacman --color auto -U /tmp/yaourt-tmp-shilkovsky/PKGDEST.oLW/php56-5.6.17-4-x86_64.pkg.tar.xz'
[2016-02-03 10:27] [ALPM] transaction started
[2016-02-03 10:27] [ALPM] installed php56 (5.6.17-4)

in pacman.log, but there is nothing actually related to building process

**Comment by f2404** [comment]
I mean the standard output - it starts with the following for me:
```
php56 5.6.17-4  (2016-01-16 19:30)
( Unsupported package: Potentially dangerous ! )
==> Edit PKGBUILD ? [Y/n] ("A" to abort)
==> ------------------------------------
==> n

==> WARNING: This PKGBUILD describes a split package.
==> Specific package options are unknown
==> php56 dependencies:
 - libldap (already installed) [makedepend]
 - sqlite (already installed) [makedepend]
 - enchant (already installed) [makedepend]
 - file (already installed) [makedepend]
 - aspell (already installed) [makedepend]
 - libltdl (already installed) [makedepend]
 - icu (already installed) [makedepend]
 - curl (already installed) [makedepend]
 - libxslt (already installed) [makedepend]
 - openssl (already installed) [makedepend]
 - db (already installed) [makedepend]
 - gmp (already installed) [makedepend]
 - systemd (already installed) [makedepend]
 - apache (package found) [makedepend]
 - c-client (package found) [makedepend]
 - postgresql-libs (package found) [makedepend]
 - smtp-forwarder (package found) [makedepend]
 - unixodbc (package found) [makedepend]
 - net-snmp (package found) [makedepend]
 - libzip (package found) [makedepend]
 - freetds (package found) [makedepend]
 - libmcrypt (package found) [makedepend]
 - tidyhtml (package found) [makedepend]
 - gd (package found) [makedepend]


==> Continue building php56 ? [Y/n]
==> -------------------------------
==> 
```

**Comment by f2404** [comment]
For example, if `gd.so` and `mcrypt.so` are missing for you, this could mean that the corresponding packages (`gd` and `libmcrypt`) haven't been built during installation.

**Comment by shilka** [comment]
I see, but the log from stdout was lost since the time i installed the package. I can try to reproduce and save a log. In time a was running everything seems ok, but there were a lot of output, maybe i missed something. The build did not stoped with any errors also (and it really should?).

**Comment by f2404** [comment]
I'm not sure if it will stop on failure or not...
BTW - which `yaourt` version are you using? Can you try the git version `yaourt-git` in case you are using the release version?

**Comment by larchunix** [comment]
@shilka: Did yaourt install php56 subpackages (`php56-gd`, `php56-mcrypt`, ...) ?

**Comment by shilka** [comment]
yaourt --version    
yaourt 1.7
Did not know about -git packet. I will try to reproduce with current version and save a log, if it fails i will try current git version.

@larchunix that was my first thought to do, but there seems to be no sence in all theese packages, as they are all poining to php56 (and all are built from the same source). You can see theirs pckbuilds to make sure. I don't know why they are even exists.

**Comment by larchunix** [comment]
Works for me:

```

...

==> Install php56 ? [Y/n]

Packages (21) php56-5.6.17-4  php56-apache-5.6.17-4  php56-cgi-5.6.17-4  php56-embed-5.6.17-4  php56-enchant-5.6.17-4  php56-fpm-5.6.17-4  php56-gd-5.6.17-4
              php56-imap-5.6.17-4  php56-intl-5.6.17-4  php56-ldap-5.6.17-4  php56-mcrypt-5.6.17-4  php56-mssql-5.6.17-4  php56-odbc-5.6.17-4
              php56-pear-5.6.17-4  php56-pgsql-5.6.17-4  php56-phpdbg-5.6.17-4  php56-pspell-5.6.17-4  php56-snmp-5.6.17-4  php56-sqlite-5.6.17-4
              php56-tidy-5.6.17-4  php56-xsl-5.6.17-4
```

**Comment by larchunix** [comment]
@shilka: I don't understand your point about subpackages.

**Comment by larchunix** [comment]
@shilka: what did you answer to this question ?

```
==> This PKGBUILD provides " php56 php56-cgi php56-apache php56-fpm php56-embed php56-phpdbg php56-pear php56-enchant php56-gd php56-imap php56-intl php56-ldap php56-mcrypt php56-mssql php56-odbc php56-pgsql php56-pspell php56-snmp php56-sqlite php56-tidy php56-xsl "
==> Install only php56 ? [Y/n]
==> --------------------------
==>
```

**Comment by shilka** [comment]
@larchunix yep, it seems that's where I missed the point. The confusing part was that the packages php56-* seems to build all the php56 from sratch, which made me think wrong. Thank you for time wasted on me
